eng/tools/GenerateFiles/Directory.Build.props.in

```diff
@@ -2,5 +2,6 @@
   <PropertyGroup>
     <DefaultNetCoreTargetFramework>${DefaultNetCoreTargetFramework}</DefaultNetCoreTargetFramework>
     <TreatWarningsAsErrors Condition="'$(BuildingInsideVisualStudio)' != 'true'">true</TreatWarningsAsErrors>
+    <LibNetHostAppPackVersion Condition=" '$(LibNetHostAppPackVersion)' == '' ">${LibNetHostAppPackVersion}</LibNetHostAppPackVersion>
```

Comment:
If variable not set, set it to the not set variable?

Reply:
Maybe this line isn't needed at all @dougbu ?  Since we set it in eng/tools/GenerateFiles/GenerateFiles.csproj explicitly?

Reply:
For a current datapoint: after a restore.cmd, this is what is in artifacts\bin\GenerateFiles\Directory.Build.props:
```
<Project>
  <PropertyGroup>
    <DefaultNetCoreTargetFramework>net7.0</DefaultNetCoreTargetFramework>
    <TreatWarningsAsErrors Condition="'$(BuildingInsideVisualStudio)' != 'true'">true</TreatWarningsAsErrors>
    <LibNetHostAppPackVersion Condition=" '$(LibNetHostAppPackVersion)' == '' ">7.0.0-preview.2.22103.2</LibNetHostAppPackVersion>
  </PropertyGroup>
</Project>
```

Reply:
So I think this change is correct, as its generating a Directory.Build.props.in and hardcoding the version, as a default if its not overriden in the actual vcxproj.

Reply:
This is a .props file i.e. it's loaded before your project. You might as well remove the `Condition` and override the value in a project file unconditionally (if you need to).

Reply:
> Maybe this line isn't needed at all @dougbu ?

To clarify: The line sets the property to the value that GenerateFiles.csproj provides only if the property isn't already set. Unless the property is set by something the .vcxproj imports fairly early on, the `Condition` should never be `true`. On the other hand, the .vcxproj files manually import the root Directory.Build.props and **do** have a chance to set a value before that import.

So, it's up to you how you handle this property. Definitely don't bother w/ the `Condition` if projects never need to override it and the core C++ stuff doesn't provide a value.

Reply:
nit: I don't see any overrides. Suggest removing the `Condition` here.